### PR TITLE
Fix for spins

### LINKED ISSUE #1048: Sorry. But, the problem was not solved. Should I reinstall pyiron in the "cmti001.bc.rzg.mpg.de"?
It seems like the pyiron has been updated recently. The code worked well before but now failed with the following issue. 
And the kernel frequently died when loading data...

```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[3], line 5
      3 pr, structure = io.load(filename,filetype)
      4 structure_spin = io.add_spin(pr,structure,'Fe','2.5','B','0')
----> 5 structure_spin.plot3d()

File /u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/pyiron_atomistics/atomistics/structure/atoms.py:1215, in Atoms.plot3d(self, mode, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera, opacity)
   1192 def plot3d(
   1193     self,
   1194     mode="NGLview",
   (...)
   1213     opacity=1.0,
   1214 ):
-> 1215     return self.visualize.plot3d(
   1216         mode=mode,
   1217         show_cell=show_cell,
   1218         show_axes=show_axes,
   1219         camera=camera,
   1220         spacefill=spacefill,
   1221         particle_size=particle_size,
   1222         select_atoms=select_atoms,
   1223         background=background,
   1224         color_scheme=color_scheme,
   1225         colors=colors,
   1226         scalar_field=scalar_field,
   1227         scalar_start=scalar_start,
   1228         scalar_end=scalar_end,
   1229         scalar_cmap=scalar_cmap,
   1230         vector_field=vector_field,
   1231         vector_color=vector_color,
   1232         magnetic_moments=magnetic_moments,
   1233         view_plane=view_plane,
   1234         distance_from_camera=distance_from_camera,
   1235         opacity=opacity,
   1236     )

File /u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/pyiron_atomistics/atomistics/structure/_visualize.py:100, in Visualize.plot3d(self, mode, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera, opacity)
     24 def plot3d(
     25     self,
     26     mode="NGLview",
   (...)
     45     opacity=1.0,
     46 ):
     47     """
     48     Plot3d relies on NGLView or plotly to visualize atomic structures. Here, we construct a string in the "protein database"
     49 
   (...)
     98         * The colour interpretation of some hex codes is weird, e.g. 'green'.
     99     """
--> 100     return plot3d(
    101         structure=self._ref_atoms,
    102         mode=mode,
    103         show_cell=show_cell,
    104         show_axes=show_axes,
    105         camera=camera,
    106         spacefill=spacefill,
    107         particle_size=particle_size,
    108         select_atoms=select_atoms,
    109         background=background,
    110         color_scheme=color_scheme,
    111         colors=colors,
    112         scalar_field=scalar_field,
    113         scalar_start=scalar_start,
    114         scalar_end=scalar_end,
    115         scalar_cmap=scalar_cmap,
    116         vector_field=vector_field,
    117         vector_color=vector_color,
    118         magnetic_moments=magnetic_moments,
    119         view_plane=view_plane,
    120         distance_from_camera=distance_from_camera,
    121         opacity=opacity,
    122     )

File /u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/structuretoolkit/visualize.py:100, in plot3d(structure, mode, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera, opacity)
     46 """
     47 Plot3d relies on NGLView or plotly to visualize atomic structures. Here, we construct a string in the "protein database"
     48 
   (...)
     97     * The colour interpretation of some hex codes is weird, e.g. 'green'.
     98 """
     99 if mode == "NGLview":
--> 100     return _plot3d(
    101         structure=structure,
    102         show_cell=show_cell,
    103         show_axes=show_axes,
    104         camera=camera,
    105         spacefill=spacefill,
    106         particle_size=particle_size,
    107         select_atoms=select_atoms,
    108         background=background,
    109         color_scheme=color_scheme,
    110         colors=colors,
    111         scalar_field=scalar_field,
    112         scalar_start=scalar_start,
    113         scalar_end=scalar_end,
    114         scalar_cmap=scalar_cmap,
    115         vector_field=vector_field,
    116         vector_color=vector_color,
    117         magnetic_moments=magnetic_moments,
    118         view_plane=view_plane,
    119         distance_from_camera=distance_from_camera,
    120     )
    121 elif mode == "plotly":
    122     return _plot3d_plotly(
    123         structure=structure,
    124         camera=camera,
   (...)
    130         opacity=opacity,
    131     )

File /u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/structuretoolkit/visualize.py:304, in _plot3d(structure, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera)
    301         vector_field = structure.get_initial_magnetic_moments()
    303 elements = structure.get_chemical_symbols()
--> 304 atomic_numbers = get_atomic_numbers(structure=structure)
    305 positions = structure.positions
    307 # If `select_atoms` was given, visualize only a subset of the `parent_basis`

File /u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/structuretoolkit/helper.py:7, in get_atomic_numbers(structure)
      6 def get_atomic_numbers(structure):
----> 7     return [atomic_numbers[el] for el in structure.get_chemical_symbols()]

File /u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/structuretoolkit/helper.py:7, in <listcomp>(.0)
      6 def get_atomic_numbers(structure):
----> 7     return [atomic_numbers[el] for el in structure.get_chemical_symbols()]

KeyError: 'Fe_spin'
```

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

**Comment by jan-janssen** [comment]
Waiting for https://github.com/pyiron/pyiron_atomistics/pull/1056

File: pyiron_atomistics/atomistics/structure/atoms.py [hunk 1]
```diff
@@ -2484,7 +2488,7 @@ def get_initial_magnetic_moments(self):
                         ]
                     )
                 elif isinstance(spin_lst, (list, np.ndarray)):
```
Comment (samwaseda) [inline pyiron_atomistics/atomistics/structure/atoms.py:2490]:
```suggestion
```

File: pyiron_atomistics/atomistics/structure/atoms.py [hunk 2]
```diff
@@ -2484,7 +2488,7 @@ def get_initial_magnetic_moments(self):
...
-                    return np.array(spin_lst)
+                    return np.array([float(s) if s else 0.0 for s in spin_lst])
```
Comment (samwaseda) [inline pyiron_atomistics/atomistics/structure/atoms.py:2491]:
```suggestion
```

**Review by samwaseda — COMMENTED** [review]
These changes are very much compatible with the current pyiron implementation, but from my point of view, we should remove the possibility of mixing magnetic and non-magnetic atoms, because that's not only physically meaningless, but also there's no code which can handle anything like that.

**Comment by jan-janssen** [comment]
> These changes are very much compatible with the current pyiron implementation, but from my point of view, we should remove the possibility of mixing magnetic and non-magnetic atoms, because that's not only physically meaningless, but also there's no code which can handle anything like that.

The code sets the spin to `0.0` if no spin is defined by the user, I had the feeling that was the intention of https://github.com/pyiron/pyiron_atomistics/issues/1048

**Pull request merged**